### ISSUE TITLE
MATCH using -1 as type

### ISSUE THREAD

**Issue by AJNandi** [handsontable/hyperformula#1063]
```
MATCH(Searchcriterion; Lookuparray; Type)
```
The match function specifies a type as the 3rd argument. If you use "1", HF works correctly to match the lower bound. However, if you use "-1" it should return the upper bound. This is the behavior of Sheets and Excel. Hyperformula throws an error.

### MATCH working correctly
```
test("Match with negative match", () => {
  const engine = HyperFormula.buildFromArray(
    [["=MATCH(203, A2:A5, 1)"], ["100"], ["200"], ["300"], ["400"], ["500"]],
    {
      licenseKey: "gpl-v3",
    }
  );

  expect(engine.getCellValue({ sheet: 0, row: 0, col: 0 })).toEqual(2);
});
```
### MATCH throwing error

```
Error: Value of the formula cell is not computed.
```

```
test("Match with negative match", () => {
  const engine = HyperFormula.buildFromArray(
    [["=MATCH(203, A2:A5, -1)"], ["500"], ["400"], ["300"], ["200"], ["100"]],
    {
      licenseKey: "gpl-v3",
    }
  );

  expect(engine.getCellValue({ sheet: 0, row: 0, col: 0 })).toEqual(3);
});
```

I should note, this assumes that the data is ordered correctly. We should probably update the documentation to mention that requirement. When the data is ordered incorrectly, Google sheets will return the largest value, whereas hyperformula returns the same value whether its -1 or 1. We may want to update the runtime differences docs to reflect that difference if it is intended. See below for that example. You can see Google sheets explain the data ordering assumption here: https://support.google.com/docs/answer/3093378

```
test("Match with negative match", () => {
  const engine = HyperFormula.buildFromArray(
    [["=MATCH(203, A2:A5, -1)"], ["100"], ["200"], ["300"], ["400"], ["500"]],
    {
      licenseKey: "gpl-v3",
    }
  );

  expect(engine.getCellValue({ sheet: 0, row: 0, col: 0 })).toEqual(5); // Google sheets returns 5 in this scenario, hyperformula returns 2. 
});
```

**Comment by warpech** [comment]
Thank you for reporting, @AJNandi. You are right that the current implementation does not work as intended with `-1` as the third parameter. We will fix that.

Also, as you suggest, the docs should be improved to better explain the third parameter and the expected order of the data.

**Comment by sequba** [comment]
Hi @AJNandi 

I'm more than happy to announce that we just released [HyperFormula 2.2.0](https://github.com/handsontable/hyperformula/releases/tag/2.2.0) where this issue is fixed. We are closing this issue as solved. If there is anything that won't work for you after updating please leave a comment.
